gems/aws-sdk-core/lib/aws-sdk-core/credential_provider_chain.rb

```diff
@@ -186,5 +195,9 @@ def assume_role_with_profile(options, profile_name)
       end
       Aws.shared_config.assume_role_credentials_from_config(assume_opts)
     end
+
+    def with_metrics(metric, &block)
```

Comment:
If there's no specific logic happening with in `with_metric`, may not need this helper function. I.e: [using class method directly](https://github.com/aws/aws-sdk-ruby/blob/version-3/gems/aws-sdk-s3/lib/aws-sdk-s3/customizations/object.rb#L79) vs [specific logic](https://github.com/aws/aws-sdk-ruby/blob/version-3/gems/aws-sdk-core/lib/aws-sdk-core/plugins/retry_errors.rb#L384-L386)